### DISCUSSION TITLE
Autocomplete for context does not work in resolver

### DISCUSSION THREAD

**Discussion by sfratini** [prisma/prisma#3097] [Q&A] [answered]
Hi! I have a a few models that require a secondary table to get data. These are all models that support multiple translations so I want to require a parameter to read/save data. As I am writing the field for the name, which requires the query, I see that the ctx does not have autocomplete (however the parent variable works). Is there something am I missing?

```ts
export const Category = objectType({
  name: 'categories',
  definition(t){
    t.model.id()
    t.field("name", {
      type: 'String',
      args: {
        lang: arg({type: "String", default: "en_US", required: true})
      },
      resolve(parent, {lang}, ctx){
        return ctx.db. //Here I get not suggestions
      }
    })
    t.model.created()
    t.model.companies()
  }
})
```

My schema.ts
```ts
import { makeSchema } from '@nexus/schema';
import { nexusSchemaPrisma } from 'nexus-plugin-prisma/schema';
import * as graphqlTypes from './graphql';
import * as path from 'path'

export const schema = makeSchema({

    shouldGenerateArtifacts: true,

    types: graphqlTypes,

    plugins: [
      nexusSchemaPrisma({
          experimentalCRUD: true
      })
    ],

    typegenAutoConfig: {
        sources: [
          {
            source: '@prisma/client',
            alias: 'client'
          }
        ]
      },
 
    outputs: {
        schema: path.join(__dirname, './schema.graphql'),
        typegen: path.join(__dirname, './typegen.ts')
    }

});
```

And my tsconfig (it is not a new project, I already had a react web here)
```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react"
  },
  "include": [
    "src",
    "api/**/*",
    "types.d.ts"
  ]
}
```

Also, while a bit unrelated, I'd appreciate if someone can also point me how to implement the mutation in this case, as I have to write one record an then tied them together. But that is something I'll deal with later. 

Thanks!

**Comment by ryands17** [comment, marked as answer]
Hey @sfratini 👋 
Create your context file like this *context.ts*

```ts
import { PrismaClient } from '@prisma/client'

export interface Context {
  db: PrismaClient
}
```

Your *makeSchema* method should look something like this: 

```ts
export const schema = makeSchema({

    shouldGenerateArtifacts: true,

    types: graphqlTypes,

    plugins: [
      nexusSchemaPrisma({
          experimentalCRUD: true
      })
    ],

    typegenAutoConfig: {
       contextType: 'Context.Context',
        sources: [
          {
            source: '@prisma/client',
            alias: 'client'
          },
        {
          source: require.resolve('./context'),
          alias: 'Context',
        },
        ]
      },
 
    outputs: {
        schema: path.join(__dirname, './schema.graphql'),
        typegen: path.join(__dirname, './typegen.ts')
    }
});
```

Restart VSCode if required.
So now when you type ctx.db. it should give you autocompletion based on the Prisma Client types.

For reference, you can check out [this](https://github.com/ryands17/nexus-auth/) repo as well.

**Reply by sfratini** [reply]
Excellent! That worked. I was reading the docs and I assumed the db was added automatically. Is it correct to assume the framework does it automatically but the library does not? Thanks again!

**Reply by ryands17** [reply]
Yes you assume correctly :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
